crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -1156,6 +1188,214 @@ impl<'src> CodeExampleRst<'src> {
     }
 }
 
+/// Represents a code example extracted from a Markdown [fenced code block].
+///
+/// [fenced code block]: https://spec.commonmark.org/0.30/#fenced-code-blocks
+#[derive(Debug)]
+struct CodeExampleMarkdown<'src> {
+    /// The lines that have been seen so far that make up the block.
+    lines: Vec<CodeExampleLine<'src>>,
+
+    /// The indent of the line "opening" fence of this block measured via
+    /// `indentation_length`.
+    ///
+    /// This indentation is trimmed from the indentation of every line in the
+    /// body of the code block,
+    opening_fence_indent: TextSize,
+
+    /// The kind of fence, backticks or tildes, used for this block. We need to
+    /// keep track of which kind was used to open the block in order to look
+    /// for a correct close of the block.
+    fence_kind: MarkdownFenceKind,
+
+    /// The size of the fence, in codepoints, in the opening line. A correct
+    /// close of the fence must use *at least* this many characters.
+    fence_len: usize,
```

Comment:
That's a lot of supported ticks 🤣 But makes sense, using `u32` would require some annoying casts

Reply:
Yeah I did consider this when I wrote down the data type, but didn't see a good reason to artificially restrict it. That is, I don't any issues arising from a lot of ticks that we would be unique from the more general issue of a supremely long line.

But if we wanted to restrict this, probably `u8` would be good enough. 255 ticks ought to be enough for anyone, right?